specification/vi/resource-manager/Microsoft.VideoIndexer/preview/2021-04-01-preview/vi.json

```diff
@@ -0,0 +1,752 @@
+        "name": {
+          "description": "The name of the metric.",
+          "type": "string"
+        },
+        "displayName": {
+          "description": "Localized display name of the metric.",
+          "type": "string"
+        },
+        "displayDescription": {
+          "description": "The description of the metric.",
+          "type": "string"
+        },
+        "unit": {
+          "description": "The unit that the metric is measured in.",
+          "type": "string"
+        },
+        "aggregationType": {
+          "description": "The type of metric aggregation.",
+          "type": "string"
+        },
+        "enableRegionalMdmAccount": {
+          "description": "Whether or not the service is using regional MDM accounts.",
+          "type": "string"
+        },
+        "sourceMdmAccount": {
+          "description": "The name of the MDM account.",
+          "type": "string"
+        },
+        "sourceMdmNamespace": {
+          "description": "The name of the MDM namespace.",
+          "type": "string"
+        },
+        "availabilities": {
+          "description": "Defines how often data for metrics becomes available.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/OperationMetricAvailability"
+          }
+        }
+      }
+    },
+    "OperationMetricAvailability": {
+      "description": "Defines how often data for a metric becomes available.",
+      "type": "object",
+      "properties": {
+        "timeGrain": {
+          "description": "The granularity for the metric.",
+          "type": "string"
+        },
+        "blobDuration": {
+          "description": "Blob created in the customer storage account, per hour.",
+          "type": "string"
+        }
+      }
+    },
+    "OperationListResult": {
+      "description": "Available operations of the service.",
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/Operation"
+          },
+          "description": "List of operations supported by the Resource Provider."
+        },
+        "nextLink": {
+          "type": "string",
+          "description": "URL to get the next set of operation list results if there are any.",
+          "example": "https://management.azure.com/{operation}?api-version={version}&%24skiptoken={token}"
+        }
+      }
+    },
+    "AccessToken": {
+      "description": "Object that includes an account's access token.",
+      "type": "object",
+      "properties": {
+        "accessToken": {
+          "type": "string",
+          "readOnly": true,
+          "description": "The account access token.",
+          "x-ms-secret": true
+        }
+      }
+    },
+    "ErrorResponse": {
+      "description": "Error response.",
+      "properties": {
+        "error": {
+          "$ref": "#/definitions/ErrorDefinition",
+          "description": "The error details."
+        }
+      }
```

Comment:
In [ARM wiki ](https://armwiki.azurewebsites.net/rpaas/rpaasextensionapi.html#resourcecreationvalidate)I read that: 

> A response is expected under 1 minute in the form of an Azure operation result aka 200 OK with status and optional error property. Any other responses like 4xx or 5xx will be ignored to avoid issues (e.g. if an RP was not able to understand preflight request and returned 4xx or 5xx). Note, the only way to fail a preflight request (and terminate deployment) return 200 OK with Failed status and error message.
> 
> 404: returned by MetaRP if resource group or subscription not found. ResourceCreationBegin will not be called in this case.
> Response Body
> 
> Sample response body with status and error property. The message in "error" property will be returned to the customers. The format of "error" matches the OData v4.0 schema.
> 

```
{​ 
  "status": "failed",
  "error": {​ 
      "Code": "error code", 
      "Message": "error message" 
  }​ 
}​
```
does that mean that there is no reason to specify the list of error codes (401, for instance) my server can return?  

Reply:
For SDK part, you do not need to specify every error codes just using default.

Reply:
and for non SDK?

Reply:
For non SDK, if you want to define some non-200 status code, pls take a look at this doc https://github.com/Azure/autorest/blob/master/docs/extensions/readme.md#x-ms-error-response.

Reply:
In this doc: https://armwiki.azurewebsites.net/rpaas/rpaasextensionapi.html#extensions-available-for-put it says: 

>  Any other responses like 4xx or 5xx will be ignored to avoid issues 
meaning there is no value adding responses other than default. no?

Reply:
Yes. Generally you don't need to specific 4xx or 5xx error, only if you have some bussiness logic on 4xx or 5xx.